### PR TITLE
Add realistic hlrc response serialization test base class

### PR DESCRIPTION
**Pull request by martijnvg**
The base class facilitates generating a server side response test instance,
that gets serialized as xcontent, which then gets parsed into a hlrc response
instance, which then gets asserted against the server side response instance.

This way of testing is more realistic then how hlrc response classes are tested
today, which basically tests that serialization works by generating
hlrc response instance, serialize that to xcontent and then parse it back
to a hlrc response instance.

Besides adding a base test class, this change also cuts AcknowledgedResponseTests
and BroadcastResponseTests over to use this base class.

Relates to #39745

### CODE REVIEW

**Comment by hub-cap** [comment]
@martijnvg you sir, rock. Thx for tackling this.

**Review by jasontedor — COMMENTED** [review]
I like the approach. Any reason this is still a draft?

**Comment by martijnvg** [comment]
> I like the approach. Any reason this is still a draft?

I wanted to add javadocs to the base class and then forgot about it.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractResponseTestCase.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            final XContentType xContentType = randomFrom(XContentType.values());
+            final BytesReference bytes = toShuffledXContent(serverTestInstance, xContentType, ToXContent.EMPTY_PARAMS, randomBoolean());
+
+            final XContent xContent = XContentFactory.xContent(xContentType);
+            final XContentParser parser = xContent.createParser(
+                new NamedXContentRegistry(ClusterModule.getNamedXWriteables()),
+                LoggingDeprecationHandler.INSTANCE,
+                bytes.streamInput());
+            final H hlrcInstance = doParseInstance(parser);
+            assertInstances(serverTestInstance, hlrcInstance);
```
Comment (jasontedor) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/AbstractResponseTestCase.java]: hlrcInstance -> clientInstsnce everywhere?

**Review by jasontedor — COMMENTED** [review]
I left some quips on the naming.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**